### PR TITLE
Send Create/Update Mutation from Request Group Form

### PR DESCRIPTION
**Pull request by meganniu**
Testing URLs:
+ http://localhost:3000/edit-request-group-form
+ http://localhost:3000/create-request-group-form

Changes:
+ server
  + createRequestGroup: creates the request types of the new request group if requestGroup.requestTypeNames is specified
  + updateRequestGroup: if requestGroup.requestTypeNames is specified, then
    + solf delete the request types from the existing request group that aren't present in requestGroup.requestTypeNames
    + create the request types from requestGroup.requestTypeNames that aren't present in the existing request group
+ Request Group Form
  + Added spinner that runs while the Edit Request Group form  is loading
  + Send graphql submissions on submit

### CODE REVIEW

File: client/src/components/organisms/RequestGroupForm.tsx [hunk 1]
```diff
@@ -250,90 +309,96 @@ const RequestGroupForm: FunctionComponent<Props> = (props: Props) => {
...
-                formItemName="Group Name"
-                errorString={nameError}
-                isDisabled={false}
-                tooltipText="Groups describe the overall category of an item, such as stroller, crib, or bed."
-                inputComponent={<TextField
-                  name="name"
-                  placeholder="Enter a group name"
-                  type="text"
-                  input={name}
+      {loadingRequestGroup
```
Comment (meganniu) [inline client/src/components/organisms/RequestGroupForm.tsx:326]: these are just indenting changes. you can ignore them

File: client/src/components/organisms/RequestGroupForm.tsx [hunk 2]
```diff
@@ -54,42 +56,94 @@ const RequestGroupForm: FunctionComponent<Props> = (props: Props) => {
...
+        setName(retrievedRequestGroup.name ? retrievedRequestGroup.name : "")
+        setDescription(retrievedRequestGroup.description ? retrievedRequestGroup.description : "")
+        setImage(retrievedRequestGroup.image ? retrievedRequestGroup.image : "")
 
         setRequestTypeNames(
-          res.requestTypes ?
-            res.requestTypes
-              .filter((requestType) => requestType.name)
+          retrievedRequestGroup.requestTypes ?
+            retrievedRequestGroup.requestTypes
```
Comment (Anand-Issac) [inline client/src/components/organisms/RequestGroupForm.tsx:132]: I saw that you added requestTypeNames to the schema in the server so I was wondering if that could be used here? Or maybe it's not necessary to add to the schema

**Review by Anand-Issac — APPROVED** [review]
Everything LGTM :)) 
Idk how you did all of this so fast holy shit lool

I was able to verify that:
- edit req group worked and was reflected in the db
- create req group worked and was reflected in the db

File: server/graphql/utils/requestGroup.ts [hunk 3]
```diff
@@ -24,7 +26,58 @@ const nextRequestRequestGroupHelper = (requestTypeIds, dataSources): RequestInte
...
+  return newRequestGroup
+}
+
 const updateRequestGroupHelper = async (requestGroup, dataSources, session): Promise<Document> => {
+  if (!requestGroup.id) {
+    throw new UserInputError('Missing argument value', { argumentName: 'id' })
+  }
+
+  if (requestGroup.requestTypeNames) {
+    const oldRequestGroup = await dataSources.requestGroups.getById(requestGroup.id)
```
Comment (kevin-wangg) [inline server/graphql/utils/requestGroup.ts:53]: I've sometimes encountered issues where nothing gets found if I don't convert the requestGroup id to a string. Not sure if you've encountered something similar if but it works for now i guess it's fine.

**Review by kevin-wangg — APPROVED** [review]
wow this looks really good!

**Review by kevin-wangg — COMMENTED** [review]
whoops

**Pull request merged**